x-pack/plugins/reporting/server/browsers/chromium/driver_factory/index.ts

```diff
@@ -0,0 +1,212 @@
+  browserTimezone: string;
+}
+
+export class HeadlessChromiumDriverFactory {
+  public type = 'chromium';
+  private binaryPath: string;
+  private logger: any;
+  private browserConfig: IBrowserConfig;
+
+  constructor(binaryPath: string, logger: any, browserConfig: IBrowserConfig) {
+    this.binaryPath = binaryPath;
+    this.logger = logger.clone(['chromium-driver-factory']);
+    this.browserConfig = browserConfig;
+  }
+
+  public test({ viewport, browserTimezone }: ILaunchArgs, log: (message: string) => any) {
+    const userDataDir = fs.mkdtempSync(path.join(os.tmpdir(), 'chromium-'));
+    const chromiumArgs = args({
+      userDataDir,
+      viewport,
+      verboseLogging: true,
+      disableSandbox: this.browserConfig.disableSandbox,
+      proxyConfig: this.browserConfig.proxy,
+    });
+
+    return puppeteer
+      .launch({
+        userDataDir,
+        executablePath: this.binaryPath,
+        ignoreHTTPSErrors: true,
+        args: chromiumArgs,
+        env: {
+          TZ: browserTimezone,
+        },
+      })
+      .catch((error: Error) => {
+        log(`Issues testing chromium launch, you may have troubles generating reports: ` + error);
+        log(`See chromium's log output in ${path.join(this.binaryPath, '..', 'chrome_debug.log')}`);
+        return null;
+      });
+  }
+
+  public create({ viewport, browserTimezone }: ILaunchArgs) {
+    return Rx.Observable.create(async (observer: Rx.Observer<any>) => {
+      const userDataDir = fs.mkdtempSync(path.join(os.tmpdir(), 'chromium-'));
+      const chromiumArgs = args({
+        userDataDir,
+        viewport,
+        verboseLogging: this.logger.isVerbose,
+        disableSandbox: this.browserConfig.disableSandbox,
+        proxyConfig: this.browserConfig.proxy,
+      });
+
+      let browser: any;
+      let page;
+      try {
+        browser = await puppeteer.launch({
+          userDataDir,
+          executablePath: this.binaryPath,
+          ignoreHTTPSErrors: true,
+          args: chromiumArgs,
+          env: {
+            TZ: browserTimezone,
+          },
+        });
+
+        page = await browser.newPage();
+      } catch (err) {
+        observer.error(new Error(`Error spawning Chromium browser: ${err}`));
+        throw err;
+      }
+
+      safeChildProcess(
+        {
+          async kill() {
+            await browser.close();
+          },
+        },
+        observer
+      );
+
+      // Register with a few useful puppeteer event handlers:
+      // https://pptr.dev/#?product=Puppeteer&version=v1.10.0&show=api-event-error
+      // https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#class-page
+
+      const stderr$ = Rx.fromEvent(page, 'console').pipe(
+        filter((line: any) => line._type === 'error'),
+        map((line: any) => line._text),
+        share()
+      );
+
+      const [consoleMessage$, message$] = partition((msg: any) =>
+        msg.match(/\[\d+\/\d+.\d+:\w+:CONSOLE\(\d+\)\]/)
+      )(stderr$);
```

Comment:
Is this supposed to be `stderr$.pipe` ?

Reply:
I thought so as well, but according to this SO answer you don't need to pass the pipe: https://stackoverflow.com/questions/50857298/how-to-use-rxjs6-partition-in-typescript